ui/lib/kv/addon/routes.js

```diff
@@ -6,14 +6,15 @@
 import buildRoutes from 'ember-engines/routes';
 
 export default buildRoutes(function () {
-  // There are two list routes because Ember won't let a query param (e.g. *path_to_secret) be blank.
+  // There are two list routes because Ember won't let a route param (e.g. *path_to_secret) be blank.
   // :path_to_secret is used when we're listing a secret directory. Example { path: '/:beep%2Fboop%2F/directory' });
   this.route('list');
   this.route('list-directory', { path: '/:path_to_secret/directory' });
   this.route('create');
   this.route('secret', { path: '/:name' }, function () {
-    this.route('details');
-    this.route('edit');
+    this.route('details', function () {
+      this.route('edit'); // route to create new version of a secret
+    });
```

Comment:
making `/edit` a child route of details gives us the ember data model of the specific _version_ we want to create a new version from (i.e. "edit"). On the details page, a dropdown gives us the option to select from different versions, when a version is selected it changes the query param and fires off a request to the backend for a particular version. 

If `/edit` remains a sibling route of `/details` then (I think) we would have to re-query for the version we want to edit when we click `Create new version`. This is because the parent route of `/secret` just queries the latest version so calling `this.modelFor('secret')` doesn't return a version-specific secret.

Reply:
This makes sense to me. Nice thought process!